### PR TITLE
oatpp: initial integration.

### PR DESCRIPTION
**Pull request by DavidKorczynski**
oatpp is an open source C++ Web Framework https://oatpp.io/. The website lists some of the customers of the project, for convenience it includes:
- https://www.eyesopen.com/
- https://www.legrand.us/
- https://milvus.io/
- https://www.radio-ip.com/en/
- https://www.icognize.de/
- https://www.deepdetect.com/

--------------------------------------
@lganzzzo would you be interested in getting oatpp into OSS-Fuzz? I did an initial integration and if you are interested, the only thing we will need from you is an email(s) that will receive bug reports from OSS-Fuzz which will be listed in the project.yaml (see the files added in this commit). I would be happy to integrate the fuzzers into the upstream repository as well.

### CODE REVIEW

**Comment by lganzzzo** [comment]
Hello @DavidKorczynski ,

Sounds good.
Please direct reports to bugs@oatpp.io

File: projects/oatpp/fuzz_mapper.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+};
+
+
+class Test1 : public oatpp::DTO {
+  DTO_INIT(Test1, DTO)
+  DTO_FIELD(String, strF);
+};
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  std::string input(reinterpret_cast<const char*>(data), size);
```
Comment (lganzzzo) [inline projects/oatpp/fuzz_mapper.cpp:33]: You can also do like this:

```cpp
oatpp::String input(reinterpret_cast<const char*>(data), size, true);
oatpp::parser::json::mapping::ObjectMapper mapper;
try {
  mapper.readFromString<oatpp::Object<Test1>>(input);
} catch(...) {}
```

In this case, only one string will be allocated

**Comment by DavidKorczynski** [comment]
@lganzzzo apologies for not mentioning this, the email need to be affiliated with a Google account in order to view detailed reports. Do you have such an email available (assuming bugs@oatpp.io is not affiliated with a google account)?

**Comment by lganzzzo** [comment]
Yep, that is bugs@oatpp.io

File: projects/oatpp/fuzz_mapper.cpp (hunk 1 above)
Comment (DavidKorczynski) [reply projects/oatpp/fuzz_mapper.cpp]: Thanks! I fixed this up. Would you be happy to get the fuzzers moved upstream so they live in the oatpp repository?

**Comment by DavidKorczynski** [comment]
Thanks for the quick replies!

File: projects/oatpp/fuzz_mapper.cpp (hunk 1 above)
Comment (lganzzzo) [reply projects/oatpp/fuzz_mapper.cpp]: > Would you be happy to get the fuzzers moved upstream so they live in the oatpp repository?

Sounds interesting. Let's try to do this.

**Review by asraa — APPROVED** [review]

**Pull request merged**